src/mpid/ch4/shm/posix/posix_coll.h

```diff
@@ -352,14 +392,30 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_allgatherv(const void *sendbuf, MPI
 
     MPIR_FUNC_ENTER;
 
-    mpi_errno = MPIR_Allgatherv_impl(sendbuf, sendcount, sendtype,
-                                     recvbuf, recvcounts, displs, recvtype, comm, errflag);
+    switch (MPIR_CVAR_ALLGATHERV_POSIX_INTRA_ALGORITHM) {
+        case MPIR_CVAR_ALLGATHERV_POSIX_INTRA_ALGORITHM_ipc_read:
+            mpi_errno = MPIDI_POSIX_mpi_allgatherv_gpu_ipc_read(sendbuf, sendcount, sendtype,
+                                                                recvbuf, recvcounts, displs,
+                                                                recvtype, comm, errflag);
+            break;
+
+        case MPIR_CVAR_ALLGATHERV_POSIX_INTRA_ALGORITHM_mpir:
+            goto fallback;
+
+        default:
+            MPIR_Assert(0);
+    }
 
     MPIR_ERR_CHECK(mpi_errno);
+    goto fn_exit;
 
-    MPIR_FUNC_EXIT;
+  fallback:
+    mpi_errno = MPIR_Allgatherv_impl(sendbuf, sendcount, sendtype,
+                                     recvbuf, recvcounts, displs, recvtype, comm, errflag);
+    MPIR_ERR_CHECK(mpi_errno);
 
   fn_exit:
+    MPIR_FUNC_EXIT;
     return mpi_errno;
   fn_fail:
     goto fn_exit;
```

Comment:
Separate the commits between adding the algorithm and using the algorithm.

Reply:
Fixed.